src/main/java/edu/ksu/canvas/impl/SubmissionImpl.java

```diff
@@ -0,0 +1,101 @@
+package edu.ksu.canvas.impl;
+
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.gson.reflect.TypeToken;
+import edu.ksu.canvas.interfaces.SubmissionReader;
+import edu.ksu.canvas.interfaces.SubmissionWriter;
+import edu.ksu.canvas.model.Progress;
+import edu.ksu.canvas.model.assignment.Submission;
+import edu.ksu.canvas.net.Response;
+import edu.ksu.canvas.net.RestClient;
+import edu.ksu.canvas.oauth.OauthToken;
+import edu.ksu.canvas.requestOptions.MultpleSubmissionsOptions;
+import org.apache.log4j.Logger;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.Optional;
+
+public class SubmissionImpl extends BaseImpl<Submission, SubmissionReader, SubmissionWriter> implements SubmissionReader, SubmissionWriter{
+    private static final Logger LOG = Logger.getLogger(SubmissionImpl.class);
+
+    /**
+     * Construct a new CanvasApi class with an OAuth token
+     *
+     * @param canvasBaseUrl      The base URL of your canvas instance
+     * @param apiVersion         The version of the Canvas API (currently 1)
+     * @param oauthToken         OAuth token to use when executing API calls
+     * @param restClient         a RestClient implementation to use when talking to Canvas
+     * @param connectTimeout     Timeout in seconds to use when connecting
+     * @param readTimeout        Timeout in seconds to use when waiting for data to come back from an open connection
+     * @param paginationPageSize How many objects to request per page on paginated requests
+     */
+    public SubmissionImpl(String canvasBaseUrl, Integer apiVersion, OauthToken oauthToken, RestClient restClient, int connectTimeout, int readTimeout, Integer paginationPageSize) {
+        super(canvasBaseUrl, apiVersion, oauthToken, restClient, connectTimeout, readTimeout, paginationPageSize);
+    }
+
+    /**
+     * Update the grading and comments on multiple student's assignment submissions in an asynchronous job. By section.
```

Comment:
You don't need to repeat the javadoc here. It is already on the method definition in the interface that this class is implementing. Same with the ByCourse method below. Keeping javadoc accurate is already hard enough. Having it copy/pasted makes this worse.